### PR TITLE
Fix api ref + missing typings

### PR DESCRIPTION
**Pull request by lukaszjedrasik**
closes #6192
### Short Description of the PR

Fixes for broken API references after release.

### CODE REVIEW

File: packages/commercetools/api-client/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "server": "server/index.js",
   "main": "lib/index.cjs.js",
   "module": "lib/index.es.js",
-  "types": "lib/index.d.ts",
+  "types": "lib/*/*/src/index.d.ts",
```
Comment (filipsobol) [inline packages/commercetools/api-client/package.json:8]: I'm afraid that's not right. After the package is released, this path will be incorrect, and typings might stop working.

File: packages/commercetools/api-client/src/index.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
+import * as cartActions from './helpers/cart/actions';
+
 export * from './fragments';
 export * from './types/Api';
 export * from './types/GraphQL';
 export * from './types/setup';
 export { isAnonymousSession, isUserSession } from './helpers/utils';
 export { createErrorHandler } from './helpers/commercetoolsLink/graphqlError';
-export * as cartActions from './helpers/cart/actions';
+export { cartActions };
```
Comment (filipsobol) [inline packages/commercetools/api-client/src/index.ts:9]: Is there a reason for this change? The previous implementation is cleaner and does the same thing.

File: packages/core/docs/Dockerfile [hunk 3]
```diff
@@ -4,9 +4,15 @@ WORKDIR /var/www
 
 COPY . .
 
-RUN npm install \
-   && npm run build
+RUN yarn install --network-concurrency 1 \
+    && yarn build:core \
+    && yarn build:ct:api-client \
+    && yarn cache clean --all
```
Comment (filipsobol) [inline packages/core/docs/Dockerfile:10]: Can you explain `--network-concurrency 1` and `yarn cache clean --all`?

File: packages/commercetools/api-client/src/index.ts (hunk 2 above)
Comment (lukaszjedrasik) [reply packages/commercetools/api-client/src/index.ts:9]: It is workaround for `api-extractor`
https://github.com/microsoft/rushstack/issues/2780

File: packages/commercetools/api-client/package.json (hunk 1 above)
Comment (lukaszjedrasik) [reply packages/commercetools/api-client/package.json]: Actually, we don't have `index.d.ts` inside `lib` folder

File: packages/core/docs/Dockerfile (hunk 3 above)
Comment (lukaszjedrasik) [reply packages/core/docs/Dockerfile]: I think that was unnecessary. Removed it.

File: packages/commercetools/api-client/package.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
   "server": "server/index.js",
   "main": "lib/index.cjs.js",
   "module": "lib/index.es.js",
-  "types": "lib/index.d.ts",
+  "types": "lib/commercetools/api-client/src/index.d.ts",
```
Comment (filipsobol) [inline packages/commercetools/api-client/package.json:8]: This still is incorrect. Once this package is built, the types are available under the `lib/index.d.ts` path

**Review by filipsobol — CHANGES_REQUESTED** [review]

File: packages/commercetools/api-client/package.json (hunk 1 above)
Comment (filipsobol) [reply packages/commercetools/api-client/package.json]: We do when you build the package using `yarn build:ct:tools`

File: packages/commercetools/api-client/package.json (hunk 4 above)
Comment (lukaszjedrasik) [reply packages/commercetools/api-client/package.json:8]: I'm not sure 
Zrzut ekranu 2021-08-16 o 11 29 44

Comment (lukaszjedrasik) [reply packages/commercetools/api-client/package.json:8]: Updated:

Zrzut ekranu 2021-08-16 o 13 57 39

**Review by filipsobol — APPROVED** [review]

**Pull request merged**